internal/tui/model_list.go

```diff
@@ -226,21 +320,48 @@ func (m ListModel) View() string {
 	}
 
 	m.list.Title = "How about one of these:"
-	return appStyle.Render(m.list.View())
+
+	if m.input.Focused() {
+		s += titleStyle.Render("Edit suggestion:") + "\n"
+		s += inputStyle.Render(m.input.View()) + "\n"
+		s += helpStyle.Render(m.help.View(m.keys))
+		return appStyle.Render(s)
+	}
+	s += appStyle.Render(m.list.View())
+
+	return s
 }
 
 type item struct {
 	suggestion string
+	title      string
 }
 
-func (m item) FilterValue() string { return m.suggestion }
-func (m item) Title() string       { return listItemStyle.Render(m.suggestion) }
-func (m item) Description() string {
-	return listItemHelpStyle.Render("git checkout -b '" + m.suggestion + "'")
+func (i item) Title() string { return i.title }
+func (i item) Description() string {
+	return "git checkout -b '" + i.title
 }
+func (i item) FilterValue() string { return i.suggestion }
+
+func (m ListModel) TogglePrefixes() ListModel {
+	m.usePrefix = !m.usePrefix
+	for i, item := range m.list.Items() {
+		m.list.SetItem(i, createListItem(item.FilterValue(), m.usePrefix))
+	}
+	return m
+}
+
```

Comment:
I can't do that because I need to be able to go back to the original value.

Reply:
> I can't override the original suggest because I need to be able to go back to the original value if you press T again

Right, that's why you wouldn't change the internal `item.suggestion` value, you would just change the behavior of `item.Title()`, depending on whether or not prefixes are enabled.

Reply:
Ok I see more complications with that approach than benefits, the way you propose we need to create a separate function because we need to do this for Title() and Description() plus the flag is in the model ` m.usePrefix` and Title() receiver only has the item. we would need to make this a global var. What are you thought @mxsdev based on this information?